### PR TITLE
docs: kep for new platform support

### PR DESCRIPTION
**Pull request by shaneutt**
The purpose of this PR is to add the first provisional KEP which describes how we want to document and support a process that allows contributors to add support for new platforms and integrations.

Resolves https://github.com/Kong/team-k8s/issues/55

Relates https://github.com/Kong/kubernetes-ingress-controller/issues/451, https://github.com/Kong/kubernetes-ingress-controller/pull/926

This iteration is intentionally being marked as on-hold and should not be acted on: we want to save the context, but we have many other priorities taking precedent.

### CODE REVIEW

**Comment by mflendrich** [comment]
I need to collect my thoughts because I have conflcited feelings: on one hand I'm all for making our platform support consistent and transparent, on the other hand we may be getting too close to [FizzBuzz Enterprise Edition](https://github.com/EnterpriseQualityCoding/FizzBuzzEnterpriseEdition).

**Comment by shaneutt** [comment]
> I need to collect my thoughts because I have conflcited feelings: on one hand I'm all for making our platform support consistent and transparent, on the other hand we may be getting too close to [FizzBuzz Enterprise Edition](https://github.com/EnterpriseQualityCoding/FizzBuzzEnterpriseEdition).

</img>

**Comment by mflendrich** [comment]
In a zoom call with @shaneutt we've identified smaller problems that are potentially solvable with this KEP:
1. Make KIC support arm64
2. Define criteria based on which KIC maintainers will be able to call a platform X or an integration with Y "supported"
3. Improve the experience for community users asking for support of platform X or an integration with Y
4. Define criteria by which KIC maintainers will decide whether to commit to supporting platform X / integration with Y

TBD: which of these motivations we want to take care of

File: railgun/keps/0006-officially-supported-platforms.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+- establish testing and infrastructure criteria
+- document the timeframes relevant to any official support, with clear end of life (EOL) criteria
+- provide walkthrough documentation that covers beginning to end
+- provide a repository issue template for this purpose which links to the relevant documentation
+- exercise and demonstrate all of the above by implementing the [previously requested ARM64 support][issues-451]
+
+[arm64]:https://en.wikipedia.org/wiki/ARM64
+[upstream]:https://github.com/kong/kong
+[issues-451]:https://github.com/Kong/kubernetes-ingress-controller/issues/451
+
```
Comment (shaneutt) [inline railgun/keps/0006-officially-supported-platforms.md:41]: > In a zoom call with @shaneutt we've identified smaller problems that are potentially solvable with this KEP:
> 
>     1. Make KIC support arm64
>     2. Use the process (being built here) to inform the decision whether we want arm64 support or not
>     3. Define criteria based on which KIC maintainers will be able to call a platform X or an integration with Y "supported"
>     4. Improve the experience for (communicate properly with) community users asking for support of platform X or an integration with Y
>     5. Define criteria by which KIC maintainers will decide whether to commit to supporting platform X / integration with Y
> 
> TBD: which of these motivations we want to take care of

In our zoom earlier we discussed how number 5 here is ultimately the contentious one and we had alignment on the remainder. To bridge the gap I suggest this change in language: 

```suggestion

# Non Goals

As a result of this proposal we want to explicitly stay away from having processes that are developed out of this become authoritative "accepted"/"declined" processes in favor of them being prescriptive to help contributors, but if we're going to come up with a process for hard declines or hard committing to features that should be a later iteration after we've had some time to see the effects of these changes.
```

Comment (shaneutt) [reply railgun/keps/0006-officially-supported-platforms.md:41]: lmkwyt @mflendrich ?

**Review by mflendrich — APPROVED** [review]

**Pull request merged**